### ISSUE TITLE
Drop Down contact list freezes

### ISSUE THREAD

**Issue by davidpr78** [nextcloud/mail#9624]
### Steps to reproduce

Write an E-Mail and choose more than one recipient.

### Expected behavior

You can choose many contacts as recipients.

### Actual behavior

If you choose several contacts as recipients the dropdown list freezes.

### Mail app version

3.6.0

### Mailserver or service

WebhostOne

### PHP engine version

None

### Web server

None

### Database

None

**Comment by ChristophWurst** [comment]
@GretaD could you check?

**Comment by admorgan** [comment]
I can also make this happen when I start typing a name and backspace. I have to reload the page to get any entries in the to: field after this happens.

**Comment by GretaD** [comment]
yes, i can reproduce.

![Screenshot from 2024-05-08 10-56-27](https://github.com/nextcloud/mail/assets/12728974/cfcbb21a-2625-44d3-a360-ff4b91455395)

**Comment by StefInP** [comment]
If it can help : 
Same issue for me but not necessary for several recipients. It happens with the first recipent in the to: field.
I tried with two différent user profils, one with few contacts in the address book, an other with many contacts.
I was able to reproduce only with the second profil.
My config : Mail app 3.7.2 with NC 29.0.3 (AIO 9.1.0)

**Comment by LostinSpacetime** [comment]
I tested with several accounts with the same result. It's 100% reproducible (Mail 3.7.4). We have many contacts in our address book. (We are a school.)

**Comment by GretaD** [comment]
> I tested with several accounts with the same result. It's 100% reproducible (Mail 3.7.4). We have many contacts in our address book. (We are a school.)

yes, we reproduce it, unfortunately, we dont know yet why its happening.

**Comment by ChristophWurst** [comment]
This is not reproducible with Nextcloud 30 and Mail v4. We can't figure out the change that fixed it, so I don't know what to backport for older releases :s

**Comment by LostinSpacetime** [comment]
> This is not reproducible with Nextcloud 30 and Mail v4. We can't figure out the change that fixed it, so I don't know what to backport for older releases :s

Then, let's hope Nextcloud 30 is released before the teachers in our school start using it again, after the summer holidays.

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule

**Comment by lvarnava** [comment]
Hi! I keep getting this error again and again.

NC 27.1.11
NC 29.0.8
Mail 3.7.9

We noticed that this happens because of auto-complete. When there are many users or many contacts in the address book, the search freezes. When composing a letter, I enter one letter in the "TO" field, and the search starts going through all 1000+ contacts. This can be seen in the nginx logs. If the search is not completed, and I continue to enter more characters, then erase a few, then bam - we hang, then only F5. Single letter autocompletion is not good for server performance and security. It would be nice to be able to configure or disable it. It is also in the main NC contacts menu, but it only shows 25 contacts at a time and the search does not go to infinity.

**Comment by lvarnava** [comment]
Here is confirmation that the problem exists. Attempts to solve it.

https://github.com/nextcloud/mail/pull/10322

https://github.com/nextcloud/mail/pull/10322/files
